docs/models.md

```diff
@@ -1,198 +1,645 @@
+values for them depend on your goals and preferences, and are likely to be found
+only through experimentation.
+
+**temperature**
+
+LLMs are fundamentally token-predicting machines. For a given sequence of tokens
+(such as the prompt) an LLM predicts, for each token in its vocabulary, the
+likelihood that the token comes next in the sequence. The temperature is a
+scaling factor by which these predictions are divided before being normalized to
+a probability between 0 and 1.
+
+Low temperature values&mdash;between 0.0 and 1.0&mdash;amplify the difference in
+likelihoods between tokens, with the result that the model will be even less
+likely to produce a token it already evaluated to be unlikely. This is often
+perceived as output that is less creative. Although 0.0 is technically not a
+valid value, many models treat it as indicating that the model should behave
+deterministically, and to only consider the single most likely token.
+
+High temperature values&mdash;those greater than 1.0&mdash;compress the
+differences in likelihoods between tokens, with the result that the model
+becomes more likely to produce tokens it had previously evaluated to be
+unlikely. This is often perceived as output that is more creative. Some model
+APIs impose a maximum temperature, often 2.0.
+
+**topP**
+
+_Top-p_ is a value between 0.0 and 1.0 that controls the number of possible
+tokens you want the model to consider, by specifying the cumulative probability
+of the tokens. For example, a value of 1.0 means to consider every possible
+token (but still take into account the probability of each token). A value of
+0.4 means to only consider the most likely tokens, whose probabilities add up to
+0.4, and to exclude the remaining tokens from consideration.
+
+**topK**
+
+_Top-k_ is an integer value that also controls the number of possible tokens you
+want the model to consider, but this time by explicitly specifying the maximum
+number of tokens. Specifying a value of 1 means that the model should behave
+deterministically.
+
+Note: The astute reader might notice that the fact that these parameters are
+specific to individual models undermines the promise of being able to swap out
+models simply by passing a different value to generate. You're correct! If you
+make use of these advanced parameters, you will want to also take a look at the
+documentation on dotprompt, which provides a way to specify a model, a prompt,
+and a model configuration together as one unit.
+
+You can experiment with the effect of these parameters on the output generated
+by different model and prompt combinations by using the Developer UI. Start the
+developer UI with the genkit start command and it will automatically load all of
+the models defined by the plugins configured in your project. You can quickly
+try different prompts and configuration values without having to repeatedly make
+these changes in code.
```

Comment:
This is a nice blurb about the Dev UI, but it's somewhat buried/hidden under the model parameter definitions. Consider bumping this out into a section that briefly describes using the Dev UI for interacting and experimenting with models.